sockets/utils/getSocketUrlParts.js

```diff
@@ -82,6 +82,15 @@ function getSocketUrlParts(resourceQuery) {
   pathname = parsedQuery.sockPath || pathname;
   port = parsedQuery.sockPort || port;
 
+  if (!hostname || !pathname || !port) {
+    throw new Error(
+      '[React Refresh] Failed to get an URL for the socket connection. ' +
+        'This might occur when the execution script doesn’t have `src` attribute set. ' +
+        'Either specify it or use overlay options ' +
+        'https://github.com/pmmmwh/react-refresh-webpack-plugin/blob/main/docs/API.md#overlay'
```

Comment:
How does this sound?

```suggestion
    throw new Error([
      '[React Refresh] Failed to get an URL for the socket connection.',
      'This usually means that the current executed script doesn't have a `src` attribute set.',
      'You should either specify the socket path parameters under the `devServer` key in your Webpack config, or use the `overlay` option.',
      'https://github.com/pmmmwh/react-refresh-webpack-plugin/blob/main/docs/API.md#overlay'
      ].join('\n')
    )
```

Reply:
looks good to me!